tx.go

```diff
@@ -117,101 +114,98 @@ func AppendSenderToTransaction(sender *skademlia.Keypair, tx *Transaction, paren
 		}
 
 		tx.Depth++
-
-		sort.Slice(tx.ParentIDs, func(i, j int) bool {
-			return bytes.Compare(tx.ParentIDs[i][:], tx.ParentIDs[j][:]) < 0
-		})
 	}
 
 	tx.Sender = sender.PublicKey()
 	tx.SenderSignature = edwards25519.Sign(sender.PrivateKey(), tx.Marshal())
 
 	tx.rehash()
+
+	return tx
 }
 
-func (t *Transaction) rehash() {
+func (tx *Transaction) rehash() *Transaction {
```

Comment:
why this still returns pointer? since its method of transaction struct itself i guess it can mutate. otherwise it needs to return value.

Reply:
True, will remove that.